pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerStatTest.java

```diff
@@ -424,4 +424,22 @@ public void testAddBrokerLatencyStats() throws Exception {
         producer.close();
         log.info("-- Exiting {} test --", methodName);
     }
+
+    @Test
+    public void testProducerPendingQueueSizeStats() throws Exception {
+        log.info("-- Starting {} test --", methodName);
+        ProducerBuilder<byte[]> producerBuilder = pulsarClient.newProducer()
+                .topic("persistent://my-property/tp1/my-ns/my-topic1");
+
+        Producer<byte[]> producer = producerBuilder.enableBatching(false).create();
```

Comment:
What about adding @Cleanup ?

Reply:
this test closes `pulsarClient` after every test method, which will also close and clean up producers/consumers. However, I just added the cleanup annotation.